src/client/language/languageConfiguration.ts

```diff
@@ -2,19 +2,29 @@
 // Licensed under the MIT License.
 'use strict';
 
-import { IndentAction, languages } from 'vscode';
-import { PYTHON_LANGUAGE } from '../common/constants';
+import { IndentAction } from 'vscode';
 
 export const MULTILINE_SEPARATOR_INDENT_REGEX = /^(?!\s+\\)[^#\n]+\\$/;
+/**
+ * This does not handle all cases. However, it does handle nearly all usage.
+ * Here's what it does not cover:
+ * - the statement is split over multiple lines (and hence the ":" is on a different line)
+ * - the code block is inlined (after the ":")
+ * - there are multiple statements on the line (separated by semicolons)
+ * Also note that `lambda` is purposefully excluded.
+ */
+export const INCREASE_INDENT_REGEX = /^\s*(?:(?:async|class|def|elif|except|for|if|while|with)\b.*|(else|finally|try))\s*:\s*(#.*)?$/;
+export const DECREASE_INDENT_REGEX = /^\s*(?:else|finally|(?:elif|except)\b.*)\s*:\s*(#.*)?$/;
+export const OUTDENT_SINGLE_KEYWORD_REGEX = /^\s*(break|continue|pass|raise\b.*)\s*(#.*)?$/;
+/**
+ * Dedent the line following a return statement only if there is no dangling open array, tuple or dictionary before the cursor.
+ * A line with a closed array, tuple or dictionary will be dedented correctly.
+ */
+export const OUTDENT_RETURN_REGEX = /^\s*(return)\b([^[({]|([[({].*[\])}]))*(#.*)?$/;
```

Comment:
The following regex should do the trick and is a little easier to follow:

```suggestion
export const OUTDENT_RETURN_REGEX = /^\s*return\b(?:[^#]*[^#[({:])?\s*(#.*)?$/;
```

Reply:
There are a couple of tricky cases, so let's leave it simple for now, I'll open an issue to improve the `onEnterRules` for the return keyword.